db/migrate/20230709065227_separate_role_access_restrictions_for_service_templates.rb

```diff
@@ -0,0 +1,32 @@
+class SeparateRoleAccessRestrictionsForServiceTemplates < ActiveRecord::Migration[6.0]
+  class MiqUserRole < ActiveRecord::Base; end
+
+  def up
+    say_with_time("Updating MiqUserRole restictions so Service Templates match existing VMs") do
+      MiqUserRole.where(:read_only => false).where.not(:settings => nil).find_each do |role|
+        if role.settings.include?(":vms: :user_or_group")
+          role.settings << "  :service_templates: :user_or_group\n"
+        elsif role.settings.include?(":vms: :user")
+          role.settings << "  :service_templates: :user\n"
```

Comment:
Don't remember exactly what can be in `settings`, but I worry that this may go into the wrong place in settings.
If it is flat, then probably not a concern, but if there is any hierarchy in here, then could this will end up under another value?

Reply:
With `serialize :settings` I could ensure the new `:service_templates` key is nested under `:restrictions`.